Update Cart and Checkout sidebar design

Aljullu opened the pull request:
Fixes #3632.

This PR updates the Cart and Checkout sidebar so it matches the newest designs, that required making some changes to the Shipping rates control and payment methods too.

I tried to keep commits organized so every code change is explained by its commit description, but if something is still not clear, I can give a more verbose explanation. :slightly_smiling_face: 

### Screenshots

_Cart sidebar with no shipping options:_
![imatge](https://user-images.githubusercontent.com/3616980/107000287-e975f600-6787-11eb-92b4-c39ce463c593.png)

_Cart sidebar with 2 shipping rates:_
![imatge](https://user-images.githubusercontent.com/3616980/107000346-04486a80-6788-11eb-9ffc-9459050dd7f2.png)

_Checkout sidebar:_
![imatge](https://user-images.githubusercontent.com/3616980/107001503-07445a80-678a-11eb-9ebc-88964604c09b.png)

_Checkout shipping options & payment methods:_
![imatge](https://user-images.githubusercontent.com/3616980/107000430-25a95680-6788-11eb-963d-ecd6607e7cb4.png)

### How to test the changes in this Pull Request:

Go to the Cart and Checkout block and verify designs look good. Some ideas of things to try:
* Test in a situation where there are +1 shipping rate options, only 1 or when there are no available options.
* Test when there is more than one shipping package and when there is only one.
* Test in several themes.
* Test with a different browser font size.
* Verify things look good in the editor as well.
* Test mobile too.

### Changelog

> Improve design of cart and checkout sidebars.

Aljullu committed:
Fix ShippingRatesControl not honoring renderOption prop

Aljullu committed:
Update order summary design

Aljullu committed:
Updates TotalsCoupon design

Aljullu committed:
Update Cart block sidebar title design

Aljullu committed:
Make values bold and labels regular in Cart and Checkout sidebar

Aljullu committed:
Update styles of shipping rates selector and radio control

Aljullu committed:
Remove unnecessary overflow rule in Panel content

Aljullu committed:
Fix shipping options in Checkout block

Aljullu committed:
Add some Twenty Twenty One fixes

Aljullu committed:
Fix saved payment methods display

Aljullu committed:
Remove top spacing no longer necessary in order summary quantity badge

senadir commented:
Still reviewing the PR, but I found a weird spacing issue when we have recurring totals.
![image](https://user-images.githubusercontent.com/6165348/107148162-1b3ab880-6952-11eb-8335-c8660e830138.png)

Aljullu commented:
Good catch @senadir! I created a PR in Subscriptions to fix it: 3997-gh-woocommerce/woocommerce-subscriptions. I also proposed to create a new component to abstract this kind of styles so it's easier for extensions to add things to the sidebar + more difficult for us to break their styles.

senadir commented:
Was the narrow styling in recurring totals a side effect or intended behaviour? 
design:
![image](https://user-images.githubusercontent.com/6165348/107214567-5a2c4500-6a0a-11eb-8d06-45dfa031eb14.png)
storefront:
![image](https://user-images.githubusercontent.com/6165348/107214597-67493400-6a0a-11eb-8352-cf65da493a4c.png)

Aljullu commented:
> Was the narrow styling in recurring totals a side effect or intended behaviour?

@senadir side-effect of a too broad CSS selector. Should be fixed in the WC Subscriptions PR. :+1:

senadir committed:
turn Totals title to bold

opr reviewed (changes requested):
Hey nice work! It mostly looks great, but I noticed in the checkout block when an item name is long, it can sit flush with the price.
![image](https://user-images.githubusercontent.com/5656702/107235907-11ca5280-6a1d-11eb-8a60-f6ec0658003a.png)

We can fix this by adding `padding-right` to `.wc-block-components-order-summary-item__description` I think? If we do this we should probably remove it from `.wc-block-components-product-name` too since it doesn't seem to be doing anything there.

Aljullu committed:
Move right padding from product name to item description

Aljullu commented:
That's interesting, I wasn't able to reproduce that issue in Firefox but I could in Chrome. In any case, it should be fixed in 62b95a314942e86d139117c27b2820479d4b4493 with the changes you suggested.

opr reviewed (commented):
When viewing a product with a signup fee in Twenty Twenty it is indented. I will make the change needed to solve this.
![image](https://user-images.githubusercontent.com/5656702/107253633-688c5800-6a2e-11eb-916c-89a50105d0f8.png)

opr committed:
Take left margin off list items in product details

This is because in themes where a left margin is applied to li elements, the product detail/addon gets indented and it does not sit flush with the edge of the product details container. Adding this rule to the main wc-block-components-product-details  rule will stop other themes indenting <li>s

opr reviewed (approved):
@Aljullu I am approving this now - as mentioned it tested well and the bits that didn't have been addressed. When you've seen my change about TwentyTwenty and agree with it then feel free to 🚢
